relay-quotas/src/redis.rs

```diff
@@ -342,19 +422,52 @@ impl<T: GlobalLimiter> RedisRateLimiter<T> {
             return Ok(rate_limits);
         }
 
-        // We get the redis client after the global rate limiting since we don't want to hold the
+        // We get the Redis client after the global rate limiting since we don't want to hold the
         // client across await points, otherwise it might be held for too long, and we will run out
         // of connections.
         let mut connection = self.client.get_connection().await?;
-        let results: ScriptResult = invocation
+        let result: ScriptResult = invocation
             .invoke_async(&mut connection)
             .await
             .map_err(RedisError::Redis)?;
 
-        for (quota, state) in tracked_quotas.iter().zip(results.0) {
+        for (quota, state) in tracked_quotas.iter().zip(result.0) {
             if state.is_rejected {
+                // We can calculate the error by comparing how much the cache added to the
+                // quantity with remaining limit of the consumption and limit.
```

Comment:
I assume this should be "difference".
```suggestion
                // quantity with remaining difference of the consumption and limit.
```